src/Engine/SaveLoad.cpp

```diff
@@ -366,3 +372,78 @@ void SaveNewGame() {
 
     SaveGame(1, 1);
 }
+
+void QuickSaveGame() {
+    assert(pCurrentMapName != "d05.blv"); // Not Arena.
```

Comment:
I have a feeling this assert will actually trigger in arena.

Reply:
Yeah should be an if return

Reply:
Also check out this code:
```
            case UIMSG_Game_OpenSaveGameDialog: {
                if (pCurrentMapName == "d05.blv") {
                    engine->_statusBar->setEvent(LSTR_NO_SAVING_IN_ARENA);
                    pAudioPlayer->playUISound(SOUND_error);
```

You should have smth similar there.

Reply:
So, keep the assertion, just add an if in the event handler.